src/com/facebook/buck/file/HttpDownloader.java

```diff
@@ -57,6 +68,14 @@ public boolean fetch(BuckEventBus eventBus, URI uri, Path output) throws IOExcep
 
     try {
       HttpURLConnection connection = createConnection(uri);
+
+      if (authentication.isPresent()) {
+        PasswordAuthentication p = authentication.get();
+        String authStr = p.getUserName() + ":" + new String(p.getPassword());
+        String authEncoded = BaseEncoding.base64().encode(authStr.getBytes());
+        connection.addRequestProperty("Authorization", "Basic " + authEncoded);
```

Comment:
Can you add a check for the connection being HTTPS? Sending passwords in clear text sounds pretty wrong.